### ISSUE TITLE
Text classification evaluation has_labels bug!

### ISSUE THREAD

**Issue by HERIUN** [huggingface/transformers#38809]
in Trainer.prediction_step(...)
```python
    def prediction_step(
        self,
        model: nn.Module,
        inputs: dict[str, Union[torch.Tensor, Any]],
        prediction_loss_only: bool,
        ignore_keys: Optional[list[str]] = None,
    ) -> tuple[Optional[torch.Tensor], Optional[torch.Tensor], Optional[torch.Tensor]]:
        
        has_labels = False if len(self.label_names) == 0 else all(inputs.get(k) is not None for k in self.label_names)
        ...
```

Inputs has "labels" but has_labels=False... 

compute_metic dosen't work.

<img width="1111" alt="Image" src="https://github.com/user-attachments/assets/774c2483-bac9-4437-a4a3-f1a13014ee5e" />

<img width="1367" alt="Image" src="https://github.com/user-attachments/assets/9c23bf7a-6801-4a2f-8661-29b6bb7eaf36" />

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

```python
from datasets import load_from_disk, load_dataset
import numpy as np
import torch
import evaluate
from transformers import AutoTokenizer
from transformers import AutoModelForSequenceClassification, ModernBertForSequenceClassification
from transformers import TrainingArguments, Trainer
import os
import warnings
warnings.filterwarnings(action='ignore')


def main():
    device = torch.device('mps' if torch.backends.mps.is_available() else 'cuda' if torch.cuda.is_available() else 'cpu')
    print("device: ", device)
    torch.cuda.empty_cache()
    # loading dataset
    print("Loading dataset...")
    if os.path.exists("data/tokenized/wortschartz_30/"):
        datasets = load_from_disk("data/tokenized/wortschartz_30/")
    else:
        datasets = load_dataset("Flitto/wortschartz_31", token=os.environ.get("HF_TOKEN"))
        tokenizer = AutoTokenizer.from_pretrained("answerdotai/ModernBERT-base")
        tokenizer.model_max_length = 512

        def tokenize_function(examples):
            return tokenizer(examples['text'], padding="max_length", truncation=True)

        datasets = datasets.map(
            tokenize_function, 
            batched=True, 
            num_proc=16,
            )
        datasets.save_to_disk('data/tokenized/wortschartz_30')

    print("Done")

    train_size = len(datasets["train"])
    valid_size = len(datasets["validation"])

    print("Dataset size(train): ", train_size)
    print("Dataset size(validation): ", valid_size)

    metric = evaluate.load("accuracy")

    def compute_metrics(eval_pred):
        logits, labels = eval_pred
        predictions = np.argmax(logits, axis=-1)
        return metric.compute(predictions=predictions, references=labels)

    # loading base model
    model = ModernBertForSequenceClassification.from_pretrained("answerdotai/ModernBERT-base", num_labels=31).to(device)


    default_args = {
        "output_dir": "test_trainer",
        "num_train_epochs": 1,
        "log_level": "debug",
        "per_device_train_batch_size":25,
        "per_device_eval_batch_size":50,
        "label_names": datasets['train'].features['labels'].names,
        "logging_steps": 100,
        "do_train": True,
        "do_eval": True,
        "save_strategy": "steps",
        "save_total_limit" : 5,
        "load_best_model_at_end" : True,
        "eval_strategy": "steps",
        "eval_on_start": True,
        # "dataloader_num_workers": 16,
        "seed": 42,
        "warmup_ratio": 0.1,
        "fp16": True,
    }

    train_n_step = train_size / default_args["per_device_train_batch_size"]
    valid_n_step = valid_size / default_args["per_device_eval_batch_size"]

    default_args["save_steps"]= 5000 #train_n_step // 5
    default_args["eval_steps"]= 5000 #valid_n_step // 5


    training_args = TrainingArguments(**default_args)

    trainer = Trainer(
        model=model,
        args=training_args,
        train_dataset=datasets['train'],
        eval_dataset=datasets['validation'],
        compute_metrics=compute_metrics,
    )
    trainer.train()

    if training_args.eval_strategy != "no":
        metrics = trainer.evaluate()
        trainer.log_metrics("eval", metrics)
        trainer.save_metrics("eval", metrics)

    # Save and push to hub
    trainer.save_model(training_args.output_dir) # it only save adaptor, not merged model

    # if training_args.push_to_hub:
    #     trainer.push_to_hub(dataset_name=training_args.dataset_name)
    
if __name__ == "__main__":
    main()


```

### Expected behavior

has_labels=True
compute_metric works!

**Comment by Rocketknight1** [comment]
cc @sunmarc

**Comment by bvantuan** [comment]
Hi @HERIUN @Rocketknight1 @SunMarc ! According to the documentation, `label_names` is "The list of keys in your dictionary of inputs that correspond to the labels.". And the `inputs` of your script look like this:
inputs
```
{'labels': tensor([0, 0, 2, 3, 1, 1, 1, 1, 1, 0, 3, 1, 0, 0, 0, 1, 0, 3, 2, 1, 4, 0, 1, 0,
        0]), 'input_ids': tensor([[50281,   303,  5471,  ..., 50283, 50283, 50283],
        [50281,    74,  1928,  ..., 50283, 50283, 50283],
        [50281,    74,  1928,  ..., 50283, 50283, 50283],
        ...,
        [50281,    74,  1928,  ..., 50283, 50283, 50283],
        [50281,    74,   816,  ..., 50283, 50283, 50283],
        [50281,    74,  1928,  ..., 50283, 50283, 50283]]), 'attention_mask': tensor([[1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0],
        ...,
        [1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0],
        [1, 1, 1,  ..., 0, 0, 0]])}
```

So `label_names` should be set to the corresponding key in `inputs` which is `["labels"]`.
Perhaps the documentation should clarify more clearly how to use the `label_names` argument correctly.
